src/client_builder.rs

```diff
@@ -37,21 +37,26 @@ impl ClientBuilder {
         sa_key: ServiceAccountKey,
         readonly: bool,
     ) -> Result<Client, BQError> {
-        let scope = if readonly {
-            format!("{}.readonly", self.auth_base_url)
-        } else {
-            self.auth_base_url.clone()
-        };
-        let sa_auth = ServiceAccountAuthenticator::from_service_account_key(sa_key, &[&scope]).await?;
+        let scopes: Vec<String> = self
+            .auth_base_urls
+            .iter()
+            .map(|auth_base_url| {
+                if readonly {
+                    format!("{}.readonly", auth_base_url)
+                } else {
+                    auth_base_url.to_string()
+                }
+            })
+            .collect();
+        let sa_auth = ServiceAccountAuthenticator::from_service_account_key(sa_key, &scopes).await?;
```

Comment:
I cannot use &[&str] in from_service_account_key according to this code.
Any ideas? @yamakii